rs/execution_environment/src/canister_manager.rs

```diff
@@ -1122,6 +1123,42 @@ impl CanisterManager {
         ))
     }
 
+    /// Gets the metadata of the canister.
+    pub(crate) fn get_canister_metadata(
+        &self,
+        sender: PrincipalId,
+        canister: &CanisterState,
+        section_name: &str,
+    ) -> Result<CanisterMetadataResponse, CanisterManagerError> {
+        let Some(execution_state) = &canister.execution_state else {
+            return Err(CanisterManagerError::CanisterMetadataNoWasmModule {
+                canister_id: canister.canister_id(),
+            });
+        };
```

Comment:
I think something like `let execution_state = &canister.execution_state.ok_or(Err(...))?;` would be a bit more idiomatic. Similar for below.